tools/clang/unittests/SPIRV/DecorationTest.cpp

```diff
@@ -0,0 +1,54 @@
+//===- unittests/SPIRV/DecorationTest.cpp ----- Decoration tests ----------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "gmock/gmock.h"
+#include "clang/SPIRV/Decoration.h"
+#include "clang/SPIRV/SPIRVContext.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+TEST(ValidateType, ValidateSimpleDecorationUniqueness) {
```

Comment:
I typically don't really like to have `Validate` or `Check` or similar words in the test name since test are surely validating/checking something. Having them is not providing more information but takes the space. Instead, test name should be clear what's the condition and what's the expected outcome (from style guide). What about something like ```Test(Type, SameDecorationWoParameterShouldHaveSameAddress)```?

Also applies to the following test cases.